### DISCUSSION TITLE
How to install Wix Toolset for dummies

### DISCUSSION THREAD

**Discussion by ghost** [wixtoolset/issues#7640] [Questions] [answered]
Hello,

I have absolutely no idea how to install this software. My background is linux and java. I have some basic knowledge of Windows, but here, to be honest, I'm lost on how to install Wix Toolset.
I have downloaded from the git project the artifacts.zip files, that contains a lot of nuget files.
My windows server is offline, so that means I cannot use internet to install it.
I've read the getting started from wixtoolset.org, but I don't know where to find the "dotnet" thingy.

I need to use Wix Toolset, because I plan to create a java package (using jpackage command), and they tell me to download your software, so I can use tools like "light.exe" and "candle.exe".

I would have though this is a normal .msi/.exe installation tool, but after looking one hour on how to install these nugets... I'm lost.
I've googled it, and it was not really clear what I need download, and also in your discussions in github, I didn't find anything relevant... or maybe I am starting to be blind...

Could you help me, or at least give me an hint on what to do next?

Thank you very much in advance.

Regards,
Alessandro

**Comment by ghost** [comment]
I've found the following link: https://dotnet.microsoft.com/en-us/download
But when looking, which version should I use?
If I take .Net 6, I have the following download packages:

- SDK
- ASP.NET Core Runtime
- NET Desktop Runtime
- NET Runtime

which is the one, that I need to use?

**Reply by ghost** [reply, marked as answer]
Ok, I am really blind and tired today...
I need to use the SDK, as written in the Getting Starting from Wix Toolset website...

**Comment by pas059** [comment]
Hi,
you can find the tools  "light.exe" and "candle.exe" in the WiX 3.11.2 toolset that you can download here:
[https://github.com/wixtoolset/wix3/releases/tag/wix3112rtm](url)

**But, WiX 3.11 is not the latest version of Wix** (the latest is 4); i don't know how to get the version 4 of these tools (i'm not an expert), but perhaps this will be sufficient for you. On my side, i use WiX 4  from 1 week only, and only thru VS2022 using this:
[https://www.firegiant.com/wix/heatwave/](url), this helps.
Sorry I can't do anything more for you

Regards

**Reply by BMurri** [reply]
In v4, tools like `candle` and `light` were replaced by `wix` (which is analogous to `dotnet`) and in fact can be installed into `dotnet` (in the same way as other installable tools into dotnet).

The core part of v4 WiX is now distributed as an SDK via NuGet (using the new modern SDK-style of MSBuild files).

**Reply by robmen** [reply]
I cover this in detail early in the [Deployment Dojo](https://www.youtube.com/playlist?list=PLDlzbQXIs18slmqmdlS10_de_Cps-QRg6). [Episode 8](https://youtube.com/live/_DCFgd9u8EI) covers MSBuild integration. If you want to get the latest build this [highlight video](https://youtu.be/2iIjq6zt6z0) is good.
